File: tremorguard-backend/src/tremorguard_backend/schemas/device.py
"""设备 Pydantic 模型。"""

from typing import Literal

from pydantic import BaseModel, Field

DeviceStatus = Literal["unpaired", "paired", "connected", "disconnected", "low_battery"]


class DeviceBind(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    mac_address: str = Field(min_length=1, max_length=32)
    serial_number: str = Field(min_length=1, max_length=64)
    firmware_version: str = ""
    hardware_version: str = ""
    model: str = ""
    device_key: str = Field(min_length=8, max_length=128, description="设备密钥明文，仅存哈希")
    patient_id: str | None = None
    battery_level: int = Field(default=100, ge=0, le=100)


class DeviceResponse(BaseModel):
    id: str
    user_id: str
    name: str
    mac_address: str
    serial_number: str
    firmware_version: str
    hardware_version: str
    model: str
    patient_id: str | None
    status: DeviceStatus
    battery_level: int

    model_config = {"from_attributes": True}

File: tremorguard-frontend/packages/shared-types/src/device.ts
export interface DeviceInfo {
  id: string
  clientId: string
  serverId?: string
  name: string
  macAddress: string
  serialNumber: string
  firmwareVersion: string
  hardwareVersion: string
  model: string
  patientId?: string
  status: DeviceStatus
  pairedAt?: number
  lastConnectedAt?: number
  batteryLevel: number
  createdAt: number
  updatedAt: number
  syncStatus: 'synced' | 'pending' | 'conflict'
}

export type DeviceStatus = 'unpaired' | 'paired' | 'connected' | 'disconnected' | 'low_battery'

export interface DeviceBond {
  deviceId: string
  patientId: string
  bondedAt: number
  unbondedAt?: number
  isActive: boolean
}

export interface BleServiceUUIDs {
  tremorService: string
  deviceInfoService: string
  batteryService: string
}

export interface BleCharacteristicUUIDs {
  tremorData: string
  tremorConfig: string
  deviceInfo: string
  batteryLevel: string
  heartbeat: string
}

export const BLE_UUIDS: { services: BleServiceUUIDs; characteristics: BleCharacteristicUUIDs } = {
  services: {
    tremorService: '0000ffe0-0000-1000-8000-00805f9b34fb',
    deviceInfoService: '0000180a-0000-1000-8000-00805f9b34fb',
    batteryService: '0000180f-0000-1000-8000-00805f9b34fb',
  },
  characteristics: {
    tremorData: '0000ffe1-0000-1000-8000-00805f9b34fb',
    tremorConfig: '0000ffe2-0000-1000-8000-00805f9b34fb',
    deviceInfo: '00002a29-0000-1000-8000-00805f9b34fb',
    batteryLevel: '00002a19-0000-1000-8000-00805f9b34fb',
    heartbeat: '0000ffe3-0000-1000-8000-00805f9b34fb',
  },
}
